lib/chained_job/middleware.rb

```diff
@@ -22,7 +23,7 @@ def perform(args = {}, worker_id = nil, tag = nil)
     end
 
     def arguments_array
-      options = { job_class: self.class }
+      options = { job_class: self.class, args: @args }
```

Comment:
Instead of using an instance variable, I think we should pass a parameter instead. This way, we avoid temporal coupling between `#arguments_array` and `#perform`.